### ISSUE TITLE
Incompatibility with Magento 2.2.*

### ISSUE THREAD

**Issue by rps29** [EliasKotlyar/Twinsen_DeployHelper#3]
Twinsen_DeployHelper does not support Magento major version 2.2.*

# Issue

`class SqliteConnectionAdapter extends \Magento\Framework\Model\ResourceModel\Type\Db implements ConnectionAdapterInterface`

In Magento 2.1.* to 2.2.*, the magento framework interface method `\Magento\Framework\App\ResourceConnection\ConnectionAdapterInterface::getConnection` was changed.

# Suggestion

Update module version 1.1.0 with the adjusted SqliteConnectionAdapter::getConnection(); , and require Magento major version 2.2.* for it.

### Needed Changes
```
\Twinsen\DeployHelper\Model\Type\Db\Pdo\SqliteConnectionAdapter::getConnection(DB\LoggerInterface $logger)
```
needs to be changed to:
```
\Twinsen\DeployHelper\Model\Type\Db\Pdo\SqliteConnectionAdapter::getConnection(DB\LoggerInterface $logger = null, \Magento\Framework\DB\SelectFactory $selectFactory = null)
```

### Error when running `php bin/magento setup:di:compile`

Fatal error: Declaration of Twinsen\DeployHelper\Model\Type\Db\Pdo\SqliteConnectionAdapter::getConnection(Magento\Framework\DB\LoggerInterface $logger) must be compatible with Magento\Framework\App\ResourceConnection\ConnectionAdapterInterface::getConnection(Magento\Framework\DB\LoggerInterface $logger = NULL, Magento\Framework\DB\Selec
tFactory $selectFactory = NULL) in /var/www/html/vendor/twinsen/deployhelper/Model/Type/Db/Pdo/SqliteConnectionAdapter.php on line 20

Call Stack:
    0.0006     356576   1. {main}() /var/www/html/bin/magento:0
    4.2707   45645080   2. Magento\Framework\Console\Cli->run() /var/www/html/bin/magento:23
    4.2771   46126104   3. Magento\Framework\Console\Cli->doRun() /var/www/html/vendor/symfony/console/Application.php:117
    4.2771   46126104   4. Magento\Framework\Console\Cli->doRun() /var/www/html/vendor/magento/framework/Console/Cli.php:104
    5.1551   51585728   5. Magento\Framework\Console\Cli->doRunCommand() /var/www/html/vendor/symfony/console/Application.php:193
    5.1551   51585728   6. Magento\Setup\Console\Command\DiCompileCommand->run() /var/www/html/vendor/symfony/console/Application.php:843
    5.1555   51590552   7. Magento\Setup\Console\Command\DiCompileCommand->execute() /var/www/html/vendor/symfony/console/Command/Command.php:242
   13.9379   73939960   8. Magento\Setup\Module\Di\App\Task\Manager->process() /var/www/html/setup/src/Magento/Setup/Console/Command/DiCompileCommand.php:205
   14.4819   78264576   9. Magento\Setup\Module\Di\App\Task\Operation\RepositoryGenerator->doOperation() /var/www/html/setup/src/Magento/Setup/Module/Di/App/Task/Manager.php:56
   27.6596  202636328  10. Magento\Setup\Module\Di\Code\Reader\ClassesScanner->getList() /var/www/html/setup/src/Magento/Setup/Module/Di/App/Task/Operation/RepositoryGenerator.php:61
   27.6596  202645336  11. Magento\Setup\Module\Di\Code\Reader\ClassesScanner->extract() /var/www/html/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php:79
   27.7166  203448024  12. Magento\Setup\Module\Di\Code\Reader\ClassesScanner->includeClasses() /var/www/html/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php:108
   27.7166  203448024  13. class_exists() /var/www/html/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php:122
   27.7166  203448120  14. spl_autoload_call() /var/www/html/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php:122
   27.7166  203448296  15. Composer\Autoload\ClassLoader->loadClass() /var/www/html/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php:122
   27.7166  203448424  16. Composer\Autoload\includeFile() /var/www/html/vendor/composer/ClassLoader.php:301
   27.7168  203456472  17. include('/var/www/html/vendor/twinsen/deployhelper/Model/Type/Db/Pdo/SqliteConnectionAdapter.php') /var/www/html/vendor/composer/ClassLoader.php:414

**Comment by EliasKotlyar** [comment]
Hello @rps29 ,

I am really excited that someone is still using it with a newer Magento Version. The main idea of this extension(deploying without the database) was already implemented in the Magento-Core. I can update the extension as you suggested, but dont see any benefit of doing so.  

Can you please explain the reasons , why you prefer to use this extension over the Magento Core solution?

Greatings
Elias

**Comment by rps29** [comment]
Thank you for the quick respond @EliasKotlyar !

Forget about the issue I opened, please. You can close it. I was just updating an Magento 2 (ce) project from version 2.1.11 to version 2.2.3 - And there occurred a lot of errors for me. This was one of the errors - and I reportet it because i was unable to avoid loading the modules original class entirely (through DI stuff).

Thank you very, very much for mentioning that the functionality this module provides was already implemented in the Magento-Core. I just removed the module from my composer.json and don't miss anything.

"The reasons why i prefer to use this extension over the Magento Core solution" - I didn't spend time in figuring out what this module is used for to be honest ;) Just tryed to fix code breaking changes of the Magento 2.1.x to 2.2.x update.

Thank you in advance,
Roland
